trieste/models/model_interfaces.py

```diff
@@ -37,6 +37,17 @@ def predict(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, Tenso
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def predict_full_cov(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, TensorType]:
+        """
+        Return the predicted mean and full covariance of the latent function(s) at the specified
+        ``query_points``.
+
+        :param query_points: The points at which to make predictions.
+        :return: The predicted mean and full covariance.
+        """
+        raise NotImplementedError
```

Comment:
I don't think that all probabilistic models can provide a full_cov estimate. But maybe that's OK for now since we don't have a counter-example?

Reply:
hmmm ... that's awkward. That's another reason to make models generic (which would mean you could specify only GPflow models

Reply:
it would also be solved really nicely by my new model interface

Reply:
it's OK I guess, we'll just have to admit we don't support prob models that can't, but I'm ok to add that as a feature later

Reply:
and not all models have latent models either. This is awkward ...